lua/nvim-treesitter/highlight.lua

```diff
@@ -11,7 +13,9 @@ function M.setup(bufnr, ft)
   local query = queries.get_query(ft, "highlights")
   if not query then return end
 
-  ts.TSHighlighter.new(query, buf, ft)
+  if not M.highlighters[buf] then M.highlighters[buf] = {} end
+
+  M.highlighters[buf][ft] = ts.TSHighlighter.new(query, buf, ft)
```

Comment:
not too sure about this. Can a buffer hold more than one filetype ?
Also, how would we handle files with multiple languages (like js in html or lua in vimscript ?)

Reply:
Treesitter advice is marking special nodes as languages injections, and using mutliple parsers on different ranges of the file. For example in a Markdown file, code blocks will be marked `@injection` or whatever and then highlighting will handle this by loading and using an different parser to add highlighting inside this node.

Reply:
ok so i guess it is not yet supported by neovim's api. What about the multiple filetypes for a buffer then ?

Reply:
Actually, I do think that it is possible, for example, by applying two highlighing from two different highlighters, it might be possible to have this kind of highlighting.

Reply:
TSHighlighter would parse the file two times with two different parsers ? it doesn't seem quite right.

Reply:
the only other option is to find a way to force TSHighlighter to only parse between the desired lines/byte offsets

Reply:
i think we shouldn't think about that right now as it's not yet supported by neovim's api. We might just bring that feature to the core neovim api when we have more insight on the api.

Reply:
SGTM, I've made the requested changes